### PR TITLE
Fix rebuild full stale mirror artifacts

### LINKED ISSUE #215: rebuild full reuses stale mirror artifacts without manifest.json
## Summary

`npm run rebuild:full` can fail during install when it reuses a previously mirrored raw artifact that predates the `manifest.json` mirror-file manifest format.

## Reproduction

On `main` at `b06377a` / `v1.0.7`, run:

```powershell
npm run rebuild:full
```

Observed failure:

```text
Mirror acquire complete: 7/7 artifacts under C:/Projects/agent-harness/mirror
Error: Mirror artifact is missing manifest.json: C:/Projects/agent-harness/mirror/raw/sha256-884ba9d42562179c99a8f1561b0d6f93aaf052ceec21c1384eba651f1837a901
    at verifyMirrorFileManifest (.../dist/install/bundle.js:132:15)
```

The raw artifact directory exists and contains mirrored files such as `asset.json`, `content.txt`, and `SKILL.md`, but lacks `manifest.json`.

## Expected behavior

A full rebuild should not die on stale mirror cache layout. It should either:

- refresh/reacquire selected mirror entries whose raw artifact is missing `manifest.json`, or
- clean invalid mirror raw/index state before install.

## Notes

This surfaced immediately after cutting `v1.0.7`, so it should be fixed on a patch branch before relying on the release workflow from existing workspaces.

### PR DESCRIPTION
**Pull request by ar27111994**
## Summary

- bump package metadata to `1.0.8`
- update the changelog with the rebuild-full stale mirror cache fix
- refresh selected mirror artifacts when their raw cache directory is missing `manifest.json`
- add regression coverage for legacy mirror artifacts and preserve valid mirrored-overlap behavior

## Verification

- `npm run build && node --test dist/tests/mirror-acquire-state.test.js`
- `npm run rebuild:full`
- `npm run test:coverage && npm run coverage:gaps` — 688/688 tests, 100% statements/branches/functions/lines
- `npm run validate`

Fixes #215

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed `rebuild full` command to properly refresh mirror artifacts whose cache predates the new manifest format, preventing installation failures when upgrading existing workspaces with stale mirror index entries.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ar27111994/agent-harness/pull/216?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<h3>Greptile Summary</h3>

This PR fixes `rebuild full` for workspaces whose mirror cache predates the `manifest.json` format: stale entries are now re-acquired instead of silently failing at install time. It also improves the efficiency of `pathLooksReadable` and hardens test URL matching to address CodeQL alerts.

- **Stale-manifest detection** (`src/mirror/acquire.ts`): introduces `findAssetIdsMissingMirrorManifests`, which is called only when a selective (non-full) refresh is running; it stat-checks each eligible index entry's raw cache directory and flags any entry missing `manifest.json` for re-acquisition.
- **`pathLooksReadable` refactor**: replaces the old read-to-check-readability approach with a `stat` + `access` check, eliminating unnecessary file I/O.
- **Test updates**: adds regression coverage for the legacy-manifest scenario and the directory-as-manifest edge case; updates the \"mirrored overlap\" test to first materialise a proper mirror entry before simulating the overlap state; replaces `url.includes(host)` substring checks with proper URL hostname comparison helpers.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the change is narrowly scoped to the stale-manifest detection path and does not affect full-refresh or normal incremental-acquire flows.

The new findAssetIdsMissingMirrorManifests function is correctly short-circuited for full refreshes, integrates cleanly into the existing entry filter, and uses the same assetId key that the rest of the pipeline uses. The pathLooksReadable refactor eliminates unnecessary file reads without changing observable behaviour.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/mirror/acquire.ts | Introduces findAssetIdsMissingMirrorManifests (short-circuited for full refreshes) and integrates stale-entry detection into the entry filter; refactors pathLooksReadable to use stat+access instead of reading file content; exports the new function via mirrorAcquireInternals. |
| src/tests/mirror-acquire-state.test.ts | Adds three new regression tests for stale-manifest detection; fixes the mirrored overlap test to create a proper materialized mirror entry (with manifest) before simulating the overlap acquire state. |
| src/tests/mirror-install-gap-coverage.test.ts | Introduces parseMockFetchUrl and hasHostname helpers; replaces all url.includes(host) substring checks with proper URL hostname comparisons to resolve CodeQL URL-sanitization alerts. |
| CHANGELOG.md | Adds v1.0.8 entry documenting the stale-manifest fix and CodeQL alert resolution. |
| package.json | Version bump from 1.0.7 to 1.0.8. |

</details>

<sub>Reviews (4): Last reviewed commit: ["Require manifest paths to be files"](https://github.com/ar27111994/agent-harness/commit/24b6725247301c832df9d8dc077f749383ed5f35) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32511714)</sub>

### CODE REVIEW

**Comment by ar27111994** [comment]
Security-alert follow-up added in `697c59c`.

Resolved CodeQL alerts:
- #14 `js/incomplete-url-substring-sanitization` at `src/tests/mirror-install-gap-coverage.test.ts:1607`
- #15 `js/incomplete-url-substring-sanitization` at `src/tests/mirror-install-gap-coverage.test.ts:1917`
- #16 `js/incomplete-url-substring-sanitization` at `src/tests/mirror-install-gap-coverage.test.ts:2190`
- #17 `js/incomplete-url-substring-sanitization` at `src/tests/mirror-install-gap-coverage.test.ts:2199`

Fix: the mock fetch/assertion code now parses URLs and checks exact hostnames before applying path-specific test behavior, instead of using substring checks on the full URL string.

Verification after the alert fix:
- `npm run build && node --test dist/tests/mirror-install-gap-coverage.test.js`
- `npm run test:coverage && npm run coverage:gaps && npm run validate` — 688/688 tests, 100% statements/branches/functions/lines

I did not manually dismiss the alerts because these are real CodeQL findings fixed in code; GitHub should mark them fixed after this branch is merged and code scanning reruns on `main`.

File: src/mirror/acquire.ts [hunk 1]
```diff
@@ -1054,6 +1061,33 @@ async function readPersistedSkippedAssets(
...
+
+    const manifestPath = join(
+      projectRoot,
+      "mirror",
+      "raw",
+      sanitizeMirrorId(entry.mirrorId),
+      "manifest.json",
+    );
+    if (!(await pathLooksReadable(manifestPath))) {
+      assetIds.add(entry.assetId);
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/mirror/acquire.ts:1088]: Resolved in `e510d8f` and kept in the latest branch state: `pathLooksReadable()` no longer reads the file contents for this probe. It now uses metadata/access checks instead, so stale-manifest detection avoids full-file reads across large mirror indexes.

File: src/mirror/acquire.ts [hunk 2]
```diff
@@ -1054,6 +1065,33 @@ async function readPersistedSkippedAssets(
...
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/mirror/acquire.ts:1088]: Resolved in `24b6725`: the manifest probe now requires the path to be a regular file via `stat(...).isFile()` before checking readability, so a `manifest.json` directory or other non-file is treated as stale and reacquired.

I also added a regression test covering a stale mirror entry whose `manifest.json` path is a directory.

Validation passed:
- `npm run build && node --test dist/tests/mirror-acquire-state.test.js dist/tests/final-coverage-polish.test.js`
- `npm run test:coverage && npm run coverage:gaps && npm run validate` — 689/689 tests, 100% statements/branches/functions/lines
